packages/kotti-ui/source/kotti-filter/components/FilterRow.vue

```diff
@@ -0,0 +1,230 @@
+				type="switch"
+			/>
+			<span v-if="showValueState" v-text="valueState" />
+		</div>
+		<ButtonLink
+			class="kt-filter-row__remove"
+			:icon="Yoco.Icon.CLOSE"
+			:isLoading="isLoading"
+			:type="Kotti.Filter.ButtonLinkType.DANGER"
+			@click="handleRemove"
+		/>
+	</KtForm>
+</template>
+
+<script lang="ts">
+import { Yoco } from '@3yourmind/yoco'
+import { computed, defineComponent } from '@vue/composition-api'
+
+import {
+	KtFieldDateRange,
+	KtFieldMultiSelect,
+	KtFieldNumber,
+	KtFieldSingleSelect,
+	KtFieldText,
+	KtFieldToggle,
+	KtForm,
+} from '../../'
+import { useTranslationNamespace } from '../../kotti-translation/hooks'
+import { Kotti } from '../../types'
+import { getOperationOptions, getValueComponent } from '../utils'
+
+import ButtonLink from './ButtonLink.vue'
+
+export default defineComponent<{
+	column: Kotti.Filter.Column
+	columnOptions: Kotti.FieldSingleSelect.Props['options']
+	isFirstItem: boolean
+	isLoading: boolean
+	value: Kotti.Filter.Filter
+}>({
+	name: 'FilterRow',
+	components: {
+		ButtonLink,
+		KtFieldDateRange,
+		KtFieldMultiSelect,
+		KtFieldNumber,
+		KtFieldSingleSelect,
+		KtFieldText,
+		KtFieldToggle,
+		KtForm,
+	},
+	props: {
+		column: {
+			default: () => ({}),
+			type: Object,
+		},
+		columnOptions: {
+			default: () => [],
+			type: Array,
+		},
+		isFirstItem: {
+			default: false,
+			type: Boolean,
+		},
+		isLoading: {
+			default: false,
+			type: Boolean,
+		},
+		value: {
+			default: () => ({
+				key: null,
+				operation: null,
+				value: null,
+			}),
+			type: Object,
+		},
+	},
+	setup(props, { emit }) {
+		const translations = useTranslationNamespace('KtFilter')
+
+		const prefix = computed<string>(() =>
+			props.isFirstItem
+				? translations.value.whereLabel
+				: translations.value.andLabel,
+		)
+		const operationOptions = computed<Kotti.FieldSingleSelect.Props['options']>(
+			() => {
+				if (!props.column.type) return []
+				return getOperationOptions(props.column.type)
+			},
+		)
+		const valueComponent = computed(() => {
+			if (!props.column.type) return 'div'
+			return getValueComponent(props.column.type)
+		})
+		const valueOptions = computed(() => {
+			if (!props.column.type) return []
+			switch (props.column.type) {
+				case Kotti.Filter.FilterType.SINGLE_ENUM:
+				case Kotti.Filter.FilterType.MULTI_ENUM:
+					return props.column.options
+				default:
+					return []
+			}
+		})
+		const isOperationSelectDisabled = computed<boolean>(
+			() => operationOptions.value.length <= 1,
+		)
+		const showValueState = computed<boolean>(
+			() => props.column.type === Kotti.Filter.FilterType.BOOLEAN,
+		)
+		const valueContainerClasses = computed(() => ({
+			'kt-filter-row__value-field': true,
+			'kt-filter-row__value-field--flex': showValueState.value,
+		}))
+
+		const valueState = computed<string | null>(() => {
+			if (props.value.value === null) return null
+			return props.value.value
+				? translations.value.enabledLabel
```

Comment:
Assuming this is for the boolean toggle thing — this may not sufficiently explain the `null` case — which was one of the reasons to introduce this